instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/ClientSpan.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.tracer;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+final class ClientSpan {
+  // Keeps track of the client span in a subtree corresponding to a client request.
+  private static final ContextKey<Span> KEY =
+      ContextKey.named("opentelemetry-trace-client-span-key");
+
+  /**
+   * Returns span of type {@link SpanKind#CLIENT} from the given context or {@code null} if not
+   * found.
+   */
+  @Nullable
+  static Span fromContextOrNull(Context context) {
+    return context.get(KEY);
+  }
+
+  static Context with(Context context, Span clientSpan) {
```

Comment:
I am not quite happy with this name. It seems like "client span with that context". Oh where are you, extension functions or infix operators!

But I don't have much better suggestions right now.

Reply:
Maybe `add` is a bit better

Reply:
> Oh where are you, extension functions or infix operators!

We can't switch to Kotlin this late in the game, can we? 😄

Reply:
Don't think so :)